packages/material-ui-lab/src/Alert/Alert.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { withStyles, lighten } from '@material-ui/core/styles';
+import Close from '../internal/svg-icons/Close';
+import { IconButton } from '@material-ui/core';
+import CheckCircle from '../internal/svg-icons/CheckCircle';
+import Warning from '../internal/svg-icons/Warning';
+import Error from '../internal/svg-icons/Error';
+import Info from '../internal/svg-icons/Info';
+import { green, orange, red } from '@material-ui/core/colors';
+
+export const styles = theme => ({
+  root: {
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderRadius: theme.shape.borderRadius,
+    position: 'relative',
+    display: 'flex',
+    alignItems: 'center',
+    padding: '1em 0',
+    maxWidth: 800,
+  },
+  alertContent: {
+    display: 'flex',
+    flexGrow: 1,
+    flexDirection: 'column',
+    padding: '0 1em',
+  },
+  startIcon: {
+    marginTop: '0.25em',
+    marginRight: '-1em',
+    padding: '0 1em',
+    alignSelf: 'flex-start',
+  },
+  closeIcon: {
+    marginTop: '-0.5em',
+    alignSelf: 'flex-start',
+  },
+  success: {
+    // THESE COLORS ARE STUBBED WHILE https://github.com/mui-org/material-ui/issues/13875 gets done
+    color: green[600],
+    borderColor: green[600],
+    backgroundColor: lighten(green[600], 0.95),
+  },
+  info: {
+    // THESE COLORS ARE STUBBED WHILE https://github.com/mui-org/material-ui/issues/13875 gets done
+    color: theme.palette.primary.main,
+    borderColor: theme.palette.primary.main,
+    backgroundColor: lighten(theme.palette.primary.main, 0.95),
+  },
+  warning: {
+    // THESE COLORS ARE STUBBED WHILE https://github.com/mui-org/material-ui/issues/13875 gets done
+    color: orange[600],
+    borderColor: orange[600],
+    backgroundColor: lighten(orange[600], 0.95),
+  },
+  error: {
+    // THESE COLORS ARE STUBBED WHILE https://github.com/mui-org/material-ui/issues/13875 gets done
+    color: red[600],
+    borderColor: red[600],
+    backgroundColor: lighten(red[600], 0.95),
+  },
+});
+
+const getDefaultIcon = (type) => {
```

Comment:
What about using an object map? We could also imagine exposing an `iconMapping` prop like we have with the Typography and variantMapping.